app/controllers/events_controller.rb

```diff
@@ -0,0 +1,51 @@
+class EventsController < ApplicationController
+  before_action :set_event, only: [:destroy]
+
+  def index
+    @events = Event.all
+    @events = Event.upcoming if params[:type] === Event::UPCOMING
+    @events = Event.past if params[:type] === Event::PAST
+    @events = @events.order('from_date DESC').includes(:attachments).page(params[:page])
+  end
+
+  def new
+    @event = Event.new
+    @event.attachments.build
+  end
+
+  def create
+    @event = Event.new(event_params)
+
+    if @event.save
+      redirect_to events_path(type: Event::UPCOMING), notice: 'Event was successfully created'
+    else
+      @event.attachments.build if @event.attachments.empty?
+
+      flash[:error] = attachment_error_messages
+      render action: 'new'
+    end
+  end
+
+  def destroy
```

Comment:
@destroy can raise 2 exceptions. record not found and sql invalid statement. consider handling these exception(s) in rescue_from controller method